### PR TITLE
Add remote filesystems to docs

### PR DESCRIPTION
**Pull request by kaushikb11**
## What does this PR do?

Add remote filesystems to docs

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda

### CODE REVIEW

File: docs/source/common/remote_fs.rst [hunk 1]
```diff
@@ -0,0 +1,50 @@
+Remote filesystems
+==================
+
+PyTorch Lightning enables working with data from a variety of filesystems, including local filesystems and several cloud storage providers.
+
+The datastores could be used to load and save checkpoints, as well as for logging.
+Working with different filesystems could be accomplished by appending a protocol like "s:/" to filepaths for writing and reading data.
```
Comment (tchaton) [inline docs/source/common/remote_fs.rst:7]:
```suggestion
PyTorch Lightning enables working with data from a variety of filesystems, including local file-system or remote ones provided by Cloud Providers such as `s3` on AWS, `gcs` on Google Cloud, or `adl` on Azure.

This applies to checkpoints, as well as for logging.
Working with different filesystems could be accomplished by appending a protocol like "s3:/" to file paths for writing and reading data.
```

File: docs/source/common/remote_fs.rst [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+
+.. code-block:: python
+
+    # `default_root_dir` is the default path used for logs and weights
+    trainer = Trainer(default_root_dir="s3://lightning/data/")
```
Comment (tchaton) [inline docs/source/common/remote_fs.rst:13]:
```suggestion
    trainer = Trainer(default_root_dir="s3://my_bucket/data/")
```

File: docs/source/common/remote_fs.rst [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    trainer.fit(model)
+
+You could pass custom paths to loggers for logging data.
+
+.. code-block:: python
+
+    from pytorch_lightning.loggers import TensorBoardLogger
+
+    logger = TensorBoardLogger(save_dir="s3://lightning/logs/")
```
Comment (tchaton) [inline docs/source/common/remote_fs.rst:22]:
```suggestion
    logger = TensorBoardLogger(save_dir="s3://my_bucket/logs/")
```

File: docs/source/common/remote_fs.rst [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+* Microsoft Azure Storage: ``adl://``, ``abfs://`` or ``az://`` - Microsoft Azure Storage, using `adlfs <https://github.com/fsspec/adlfs>`__.
+* Hadoop File System: ``hdfs://`` - Hadoop Distributed File System. This uses `PyArrow <https://arrow.apache.org/docs/python/>`__ as the backend.
+
+You could learn more about the available filesystems with:
+
+.. code-block:: python
+
+    from fsspec.registry import known_implementations
+
+    print(known_implementations)
```
Comment (tchaton) [inline docs/source/common/remote_fs.rst:50]: A reference to CheckpointIO might be good here.

Comment (kaushikb11) [reply docs/source/common/remote_fs.rst:51]: Updated!

File: docs/source/common/remote_fs.rst (hunk 1 above)
Comment (kaushikb11) [reply docs/source/common/remote_fs.rst]: Updated!

**Review by tchaton — APPROVED** [review]
LGTM !

**Review by awaelchli — APPROVED** [review]

**Review by carmocca — APPROVED** [review]

**Review by akihironitta — APPROVED** [review]
Once @\awaelchli's comment is resolved, LGTM!

File: docs/source/common/remote_fs.rst [hunk 5]
```diff
@@ -0,0 +1,54 @@
+Remote filesystems
+==================
+
+PyTorch Lightning enables working with data from a variety of filesystems, including local filesystems and several cloud storage providers
+such as ``s3`` on AWS, ``gcs`` on Google Cloud, or ``adl`` on Azure.
+
+This applies to saving and writing checkpoints, as well as for logging.
+Working with different filesystems could be accomplished by appending a protocol like "s3:/" to file paths for writing and reading data.
```
[the remark this answers is not in the document]
Comment (kaushikb11) [reply docs/source/common/remote_fs.rst]: Using "could" also implies the ability to do something. It makes sense for other instances

[the remark this answers is not in the document]
Comment (kaushikb11) [reply docs/source/common/remote_fs.rst]: Using "could" also implies the ability to do something. I think it makes sense for other instances

File: docs/source/common/remote_fs.rst [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+
+.. code-block:: python
+
+    # `default_root_dir` is the default path used for logs and weights
+    trainer = Trainer(default_root_dir="s3://my_bucket/data/")
```
[the remark this answers is not in the document]
Comment (kaushikb11) [reply docs/source/common/remote_fs.rst:14]: Updated!
